openfisca_uk/variables/benunit/benefits/computed.py

```diff
@@ -0,0 +1,348 @@
+        )
+        has_children = benunit.nb_persons(BenUnit.CHILD) > 0
+        random = np.random.rand(*has_children.shape)
+        takeup = (random < 0.89) * has_children + (random < 0.82) * (
+            1 - has_children
+        )
+        return max_(
+            0,
+            (personal_allowance - income_deduction)
+            * (benunit("income_support_reported", period) > 0)
+            * takeup,
+        )
+
+
+class child_benefit(Variable):
+    value_type = float
+    entity = BenUnit
+    label = u"label"
+    definition_period = ETERNITY
+
+    def formula(benunit, period, parameters):
+        num_children = benunit.nb_persons(BenUnit.CHILD)
+        eldest_amount = (
+            min_(num_children, 1)
+            * parameters(period).benefits.child_benefit.amount_eldest
+        )
+        additional_amount = (
+            max_(num_children - 1, 0)
+            * parameters(period).benefits.child_benefit.amount_additional
+        )
+        return eldest_amount + additional_amount
+
+
+class child_tax_credit_pre_means_test(Variable):
+    value_type = float
+    entity = BenUnit
+    label = u"Child Tax Credit amount received per year, before means testing"
+    definition_period = ETERNITY
+
+    def formula(benunit, period, parameters):
+        num_exempt_children = benunit.sum(
+            benunit.members("is_CTC_child_limit_exempt", period)
+        )
+        non_exempt_children = (
+            benunit.nb_persons(BenUnit.CHILD) - num_exempt_children
+        )
+        spaces_left = max_(0, 2 - num_exempt_children)
+        children_eligible = num_exempt_children + min_(
+            spaces_left, non_exempt_children
+        )
+        yearly_amount = (
+            parameters(period).benefits.child_tax_credit.family_element
+            + parameters(period).benefits.child_tax_credit.child_element
+            * children_eligible
+        )
+        return yearly_amount * (children_eligible > 0)
+
+
+class child_working_tax_credit_reduction(Variable):
+    value_type = float
+    entity = BenUnit
+    label = u"Child and Working Tax Credit amount reduced per week from means testing"
+    definition_period = ETERNITY
+
+    def formula(benunit, period, parameters):
+        child_tax_credit_amount = benunit(
+            "child_tax_credit_pre_means_test", period
+        )
+        working_tax_credit_amount = benunit(
+            "working_tax_credit_pre_means_test", period
+        )
+        eligible_for_both = (child_tax_credit_amount > 0) * (
+            working_tax_credit_amount > 0
+        )
+        threshold = (
+            eligible_for_both
+            * parameters(period).benefits.working_tax_credit.income_threshold
+            + (child_tax_credit_amount > 0)
+            * (1 - eligible_for_both)
+            * parameters(period).benefits.child_tax_credit.income_threshold
+        )
+        reduction = (
+            max_(0, (benunit("benunit_income", period) * 52 - threshold))
```

Comment:
is it possible to normalize all data initially such that 52 doesn't have to show up in any logic code?

Reply:
We could, but the vast majority of benefits and NI rate specification is in terms of weekly income, so we'd either need to divide by 52 somewhere else, or provide both weeklyised and yearlyised income amounts?